pyblish_royalrender/plugins/integrate_royalrender.py

```diff
@@ -0,0 +1,145 @@
+            else:
+                self.sub_element(element, key, value)
+
+    def process(self, context):
+
+        # Root element
+        root_element = Element("rrJob_submitFile")
+        root_element.attrib["syntax_version"] = "6.0"
+        self.sub_element(root_element, "DeleteXML", "1")
+
+        # Job element
+        job_count = 0
+        for instance in context:
+
+            # Only consider enabled instances
+            if not instance.data.get("publish", True):
+                continue
+
+            # skipping instance if not part of the family
+            if "royalrender" not in instance.data.get("families", []):
+                msg = "No \"royalrender\" family assigned. "
+                msg += "Skipping \"%s\"." % instance
+                self.log.debug(msg)
+                continue
+
+            job_element = self.sub_element(root_element, "Job", "")
+            self.dict_to_elements(
+                instance.data["royalrenderData"], job_element
+            )
+            job_count += 1
+
+        # Skip further processing if there are not any jobs
+        if not job_count:
+            return
+
+        # Writing to disk
+        tree_element = ElementTree(root_element)
+        self.indent(tree_element.getroot())
+
+        xml_path = os.path.join(
+            tempfile.gettempdir(),
+            "pyblish_royalrender_{0}.xml".format(uuid.uuid4())
+        )
+        with open(xml_path, "w") as f:
+            tree_element.write(f)
+
+        with open(xml_path, "r") as f:
+            self.log.debug(f.read())
+
+        # Submitting to RoyalRender
+        arguments = []
+
+        rr_root = os.environ["RR_Root"]
+
+        # Windows arguments
+        if platform.system() == "Windows":
+            arguments.append(
+                os.path.join(rr_root, "bin", "win", "rrStartLocal.exe"),
+            )
+            if "pyblishRoyalRenderUI" in context.data:
+                arguments.append("rrSubmitter.exe")
+            else:
+                arguments.append("rrSubmitterConsole.exe")
+
+        arguments.append(xml_path)
+
+        startupinfo = None
+        if hasattr(subprocess, "STARTF_USESHOWWINDOW"):
+            startupinfo = subprocess.STARTUPINFO()
+            startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+
+        proc = subprocess.Popen(
+            arguments,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            startupinfo=startupinfo
+        )
+        proc.stdin.close()
+        proc.stderr.close()
+
+        output = proc.stdout.read()
+        self.log.debug(output)
+
+
+class PyblishRoyalRenderDisplayUI(pyblish.api.ContextPlugin):
+    """ Setting plugin for displaying the submitter UI """
+
+    order = PyblishRoyalRenderIntegrate.order - 0.1
+    label = "Display Royal Render Submitter"
+    active = False
+    optional = True
+
+    def process(self, context):
+
```

Comment:
Just some style issues here, get rid of the newline, and add a newline to the end of the file.

Also get rid of the `""" Space before and after  """`